src/libcore/fmt/mod.rs

```diff
@@ -1512,23 +1512,30 @@ impl Display for bool {
     }
 }
 
+#[doc(hidden)]
+#[unstable(feature = "str_internals", issue = "0")]
+pub fn escape_str(s: &str, f: &mut Formatter) -> Result {
```

Comment:
Can this avoid being exposed? It seems like a reasonable amount that's not too bad to just copy directly into the new iterator.